core/util.go

```diff
@@ -478,6 +478,36 @@ func VerifyCSR(csr *x509.CertificateRequest) error {
 	return errors.New("Unsupported CSR signing algorithm")
 }
 
+var (
+	oidExtensionRequest = asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 9, 14}
+	oidSubjectAltName   = asn1.ObjectIdentifier{2, 5, 29, 17}
+)
+
+// ExtensionsFromCSR extracts the set of requested extensions from a CSR
+func ExtensionsFromCSR(csr *x509.CertificateRequest) map[string][]byte {
```

Comment:
Don't put this in core, put it in CA where it is used, and make it unexported. If we later find we need to share it across packages, we can abstract it out into a new package.